heartbeat/monitors/active/http/http_test.go

```diff
@@ -808,6 +810,57 @@ func mustParseURL(t *testing.T, url string) *url.URL {
 	return parsed
 }
 
+// helper that compresses some content as gzip
+func gzipBuffer(t *testing.T, toZip string) *bytes.Buffer {
+	var gzipBuffer bytes.Buffer
+	gzipWriter := gzip.NewWriter(&gzipBuffer)
+	defer gzipWriter.Close()
+	_, err := gzipWriter.Write([]byte(toZip))
+	if err != nil {
+		t.Fatal(err)
+	}
+	return &gzipBuffer
+}
+
+func TestDecodesGzip(t *testing.T) {
+	gzip := gzipBuffer(t, "TestEncodingAccept")
+
+	server := httptest.NewServer(hbtest.CustomResponseHandler(gzip.Bytes(), 200, map[string]string{
+		"Content-Encoding": "gzip",
+	}))
+	defer server.Close()
+
+	evt := sendTLSRequest(t, server.URL, false, map[string]interface{}{
+		"response.include_body": "always",
+		"check.request.headers": map[string]interface{}{"Accept-Encoding": "gzip"},
+	})
+
+	content, err := evt.Fields.GetValue("http.response.body.content")
+
+	// decode gzip content if 'Content-Encoding' header is present
+	assert.NoError(t, err)
+	assert.Exactly(t, content, "TestEncodingAccept")
+}
+
+func TestNoGzipDecodeWithoutHeader(t *testing.T) {
+	gzip := gzipBuffer(t, "TestEncodingAccept")
+
+	server := httptest.NewServer(hbtest.CustomResponseHandler(gzip.Bytes(), 200, map[string]string{}))
+	defer server.Close()
+
+	evt := sendTLSRequest(t, server.URL, false, map[string]interface{}{
+		"response.include_body": "always",
+		"check.request.headers": map[string]interface{}{"Accept-Encoding": "gzip"},
+	})
+
+	content, err := evt.Fields.GetValue("http.response.body.content")
+
+	assert.NoError(t, err)
+
+	// doesn't decode gzip text without content header
+	assert.Exactly(t, content, "\x1f\x8b\b\x00\x00\x00\x00\x00\x00\xff\nI-.q\xcdK\xceO\xc9\xccKwLNN-(\x01\x04\x00\x00\xff\xffW\xbeE\x0e\x12\x00\x00\x00")
+}
+
```

Comment:
Let's add a test for no header sent, but a gzip header is returned. I think that's what the original bug was about.

Reply:
I want to clarify this before I write the test and change the production code. Per the previous comment, do we mean the _response_ header wasn't sent, or the _request_ header was never specified?

Case 1: Heartbeat omits the `Accept-Encoding: gzip` header and the server responds with gzip anyway. Heartbeat should still decode.

Case 2: Heartbeat sends the `Accept-Encoding: gzip` header and the server responds with gzip, but fails to include the `Content-Encoding: gzip` header. Heartbeat should still decode.

In essence, this is making me feel like the presence of the response header doesn't matter at all, since the default behavior now will be to test for gzip no matter what and decode if it's detected.

Does anything about that seem inconsistent with your understanding?

Reply:
Apologies for the delay here. Yes, I meant case 1. I think that's the most important case of the bunch since that prompted the initial report here.

Reply:
Okay - understood. I'll add a third test to account for this case. The prod code should handle that already.